### DISCUSSION TITLE
Sort only selected rows

### DISCUSSION THREAD

**Discussion by daviewales** [saulpw/visidata#1851] [Ideas]
The title says it all!
It would be cool to have a command which sorts, but only for selected rows.

Come to think of it, it would also be cool to have a command to slide selected rows or columns, so you could move multiple around at once.

I might see if I can figure out a .visidatarc hack to do this unless someone beats me to it!

**Comment by geekscrapy** [comment]
I would certainly use the first suggestion, not sure on the second atm.

At face value, I doesnt seem to complicated?
1. Get selected rows
2. Sort them
3. Replace the rows at their new index?

**Comment by saulpw** [comment]
@midichef added this as an experimental feature in #2295.  With develop branch (and soon 3.1), add `import experimental.sort_selected` to visidatarc and then the `sort-selected-asc` and `sort-selected-desc` commands will be available.  If multiple people start using them on a regular basis and they seem to work without corrupting data, we can migrate into features for a future release.
